src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs

```diff
@@ -1000,6 +1026,33 @@ private Task WriteToStreamAsync(ReadOnlyMemory<byte> source)
             return _stream.WriteAsync(source);
         }
 
+        private bool TryReadNextLine(out ReadOnlySpan<byte> line)
+        {
+            var buffer = new ReadOnlySpan<byte>(_readBuffer, _readOffset, _readLength - _readOffset);
+            int length = buffer.IndexOf((byte)'\n');
+            if (length < 0)
+            {
+                if (_allowedReadLineBytes < buffer.Length)
+                {
+                    ThrowInvalidHttpResponse();
+                }
+
+                line = default;
+                return false;
+            }
+
+            int bytesConsumed = length + 1;
+            _readOffset += bytesConsumed;
```

Comment:
Similarly, can we use the Consume method you added here?

Reply:
There are a bunch of places in this file that we increment _readOffset.  I don't think we need to replace just this one with Consume, and I'm not sure there's value in replacing all of them (there will still be a bunch of other places that update _readOffset in other ways).  That method was really just intended for something outside of this code to use, rather than exposing _readOffset directly.